lib/components/primitive-components/Group/Group.ts

```diff
@@ -65,6 +55,56 @@ export class Group<Props extends z.ZodType<any, any, any> = typeof groupProps>
     }
   }
 
+  doInitialSourceRender() {
+    const { db } = this.root!
+    const { _parsedProps: props } = this
+    const source_group = db.source_group.insert({
+      name: props.name ?? `unnamedsubcircuit_${this._renderId}`,
+      subcircuit_id: this.isSubcircuit
+        ? `subcircuit_${this._renderId}`
+        : undefined,
+      is_subcircuit: this.isSubcircuit,
+    })
+    this.source_group_id = source_group.source_group_id
+  }
+
+  doInitialPcbPrimitiveRender() {
+    const { db } = this.root!
+    const { _parsedProps: props } = this
+
+    // Get all child components that have pcb_component_ids
+    const childPcbComponentsIds = this.children
+      .map((child) => child.pcb_component_id)
+      .filter((id): id is string => id !== null)
+
+    const pcb_group = db.pcb_group.insert({
+      subcircuit_id: this.isSubcircuit
+        ? `subcircuit_${this._renderId}`
+        : undefined,
+      is_subcircuit: this.isSubcircuit,
+      width: 0,
+      height: 0,
+      center: { x: 0, y: 0 },
+      pcb_component_ids: childPcbComponentsIds,
+      source_group_id: this.source_group_id!,
+    })
+    this.pcb_group_id = pcb_group.pcb_group_id
+
+    for (const pcbComponentId of childPcbComponentsIds) {
+      const currentComponent = db.pcb_smtpad.getWhere({
+        pcb_component_id: pcbComponentId,
+      })
+      if (currentComponent) {
+        db.pcb_smtpad.update(currentComponent.pcb_smtpad_id, {
+          pcb_group_id: pcb_group.pcb_group_id,
+          subcircuit_id: this.isSubcircuit
+            ? `subcircuit_${this._renderId}`
+            : undefined,
+        })
+      }
+    }
```

Comment:
@seveibar As the child primitive components are getting rendered first and then the `Group`, so updating the primitive components like `smtpad` in this case after the `Group` is getting rendered is the correct approach?

Reply:
the smtpad when it's created should just add it's subcircuit_id

Reply:
Each render function runs for children first, but render phases run for ALL components before moving on to the next render phase